wiki/mnmg/Using_Infiniband_for_MNMG.md

```diff
@@ -0,0 +1,332 @@
+
+`conda install -c conda-forge -c jakirkham/label/ucx cudatoolkit=9.2 ucx-proc=*=gpu ucx python=3.7`
+
+### From Source
+
+Install autogen if it's not already installed:
+```bash
+sudo apt-get install autogen autoconf libtool
+```
+
+Optionaly install `gdrcopy` for faster GPU-Network card data transfer: 
+
+From the [ucx wiki](https://github.com/openucx/ucx/wiki/NVIDIA-GPU-Support), `gdrcopy` can be installed, and might be necessary, to enable faster GPU-Network card data transfer.
+
+Here are the install instructions, taken from [gdrcopy github](https://github.com/NVIDIA/gdrcopy)
+```bash
+git clone https://github.com/NVIDIA/gdrcopy.git
+cd gdrcopy
+make PREFIX=$CONDA_INSTALL_PREFIX CUDA=/usr/local/cuda
+sudo ./insmod.sh
+export LD_LIBRARY_PATH=$PWD:$LD_LIBRARY_PATH
+```
+
+
+```bash
+git clone https://github.com/cjnolet/ucx-py.git
+cd ucx
+git checkout fea-ext-expose_worker_and_ep
+./autogen.sh
+mkdir build && cd build
+../configure --prefix=$CONDA_PREFIX --with-cuda=/usr/local/cuda --enable-mt --disable-cma CPPFLAGS="-I//usr/local/cuda/include"
+make -j install
+```
+
+Note: If you have installed `gdrcopy`, you can add `--with-gdrcopy=/path/to/gdrcopy` to the options in `configure`
+
+Verify with `ucx_info -d`. You should expect to see line(s) with the `rc` transport:
+
+```
+#   Transport: rc
+#
+#   Device: mlx5_0:1
+#
+#      capabilities:
+#            bandwidth: 11794.23 MB/sec
+#              latency: 600 nsec + 1 * N
+#             overhead: 75 nsec
+#            put_short: <= 124
+#            put_bcopy: <= 8K
+#            put_zcopy: <= 1G, up to 8 iov
+#  put_opt_zcopy_align: <= 512
+#        put_align_mtu: <= 4K
+#            get_bcopy: <= 8K
+#            get_zcopy: 65..1G, up to 8 iov
+#  get_opt_zcopy_align: <= 512
+#        get_align_mtu: <= 4K
+#             am_short: <= 123
+#             am_bcopy: <= 8191
+#             am_zcopy: <= 8191, up to 7 iov
+#   am_opt_zcopy_align: <= 512
+#         am_align_mtu: <= 4K
+#            am header: <= 127
+#               domain: device
+#           connection: to ep
+#             priority: 30
+#       device address: 3 bytes
+#           ep address: 4 bytes
+#       error handling: peer failure
+
+```
+
+You should also expect to see lines with `cuda_copy` transport:
+
+```
+#   Transport: cuda_copy
+#
+#   Device: cudacopy0
+#
+#      capabilities:
+#            bandwidth: 6911.00 MB/sec
+#              latency: 10000 nsec
+#             overhead: 0 nsec
+#            put_short: <= 4294967295
+#            put_zcopy: unlimited, up to 1 iov
+#  put_opt_zcopy_align: <= 1
+#        put_align_mtu: <= 1
+#            get_short: <= 4294967295
+#            get_zcopy: unlimited, up to 1 iov
+#  get_opt_zcopy_align: <= 1
+#        get_align_mtu: <= 1
+#           connection: to iface
+#             priority: 0
+#       device address: 0 bytes
+#        iface address: 8 bytes
+#       error handling: none
+```
+
```

Comment:
Need to check cuda_ipc as well. This is the transport to be used for Device to Device data transfer within a node.

Reply:
While the initial `point-to-point` example in this markdown was meant specifically to test UCX for host memory transport, I would like to include your new tests here as well, which demonstrate these device memory transports. 

What do you think?

Reply:
Sure, and I am planning to submit that in a separate PR, but if you include cuda_ipc here, this document will be more future ready :-)

Reply:
Agreed. ‘cuda_ipc’ has been added.